### ISSUE TITLE
Great job

### ISSUE THREAD

**Issue by ShawnDong98** [NeuralVFX/faceswap-autoencoder#1]
Great job, thanks for sharing.

I’ve found it for a long time.

**Comment by NeuralVFX** [comment]
Thanks! A small update will be coming soon...

**Comment by ShawnDong98** [comment]
> Thanks! A small update will be coming soon...

That's cool, I'm looking forward to it.

I have some question about the mask training when I reproduce it.

The metrics of mask training only include GAN Loss， regularize alpha loss， tv_loss,  and the perceptual loss  that I haven't added, right?

But in my implementation, it always tends to be zero, it makes me confused...

Can you explain how the mask training work?

**Comment by NeuralVFX** [comment]
These two lines of code account for all of the control you can have over the mask:

>         # Alpha Mask loss
>         self.loss_batch_dict[f'M_{which}_Loss'] = 1e-2 * torch.mean(torch.abs(fake_alpha))
> 
>         # Alpha mask variation loss
>         self.loss_batch_dict[f'MV_{which}_Loss'] = .1 * (torch.mean(n.emboss(fake_alpha, axis=2)) + \
>                                                          torch.mean(n.emboss(fake_alpha, axis=3)))

"M_Loss" can be seen as a multiplier, the lower this number is set the more bright the mask will be allowed to go, if you set this high the mask will be zero

"MV_Loss" only affects the sharpness of the mask, if you set this low then the mask will have more defined edges, and when set high the mask becomes blurrier

Here are some examples, trained for just a couple epochs:

MV_Loss multiplier set to .1
![high_mv_loss](https://user-images.githubusercontent.com/38269894/80873285-0b809c00-8c7d-11ea-8f90-926c2a959267.jpg)

MV_Loss Multiplier set to 1e-15
![low_mv_loss](https://user-images.githubusercontent.com/38269894/80873308-2eab4b80-8c7d-11ea-820c-1a170172f014.jpg)

**Comment by ShawnDong98** [comment]
> These two lines of code account for all of the control you can have over the mask:
> 
> > ```
> >     # Alpha Mask loss
> >     self.loss_batch_dict[f'M_{which}_Loss'] = 1e-2 * torch.mean(torch.abs(fake_alpha))
> > 
> >     # Alpha mask variation loss
> >     self.loss_batch_dict[f'MV_{which}_Loss'] = .1 * (torch.mean(n.emboss(fake_alpha, axis=2)) + \
> >                                                      torch.mean(n.emboss(fake_alpha, axis=3)))
> > ```
> 
> "M_Loss" can be seen as a multiplier, the lower this number is set the more bright the mask will be allowed to go, if you set this high the mask will be zero
> 
> "MV_Loss" only affects the sharpness of the mask, if you set this low then the mask will have more defined edges, and when set high the mask becomes blurrier
> 
> Here are some examples, trained for just a couple epochs:
> 
> MV_Loss multiplier set to .1
> ![high_mv_loss](https://user-images.githubusercontent.com/38269894/80873285-0b809c00-8c7d-11ea-8f90-926c2a959267.jpg)
> 
> MV_Loss Multiplier set to 1e-15
> ![low_mv_loss](https://user-images.githubusercontent.com/38269894/80873308-2eab4b80-8c7d-11ea-820c-1a170172f014.jpg)

That's it. You clear my confusion. Thanks!~
